### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by thomasjw99**

### CODE REVIEW

File: src/threadfunc.c
Comment (VarunBhavya) [inline src/threadfunc.c:1]: I have a question about lines 72-76. While I recognize that pdFALSE is returned for failed operations, I don't see its necessity in this condition. Could you please explain why you've included it here? Are you considering the if statement to be a failed operation?

File: test/test.c
Comment (VarunBhavya) [inline test/test.c:1]: An unorphaned lock should address the issue with the orphaned lock. However, we are verifying for a failed operation in the unorphaned lock function. Could you let me know if you are certain the test is passing? In line 117, you should compare it with 1, as done in line 112.

Also, in the orphaned lock function, lines 90-94 and 96-100 seem to perform the same operation. Is there any particular reason for this?

**Review by VarunBhavya — CHANGES_REQUESTED** [review]
I have added my queries and thoughts regarding the issues in two files. If you believe I have misunderstood anything, please let me know. I am open to clarifying my understanding and will approve the pull request once any concerns are addressed.

File: src/threadfunc.c
Comment (thomasjw99) [reply src/threadfunc.c:1]: I have updated it to return 0. You were right about the return pdFalse as it is a logic error not a system error where the semaphore is taken.

File: test/test.c
Comment (thomasjw99) [reply test/test.c:1]: As we discussed, I have made the comment on the redundant test.

**Comment by VarunBhavya** [comment]
All the changes have been successfully implemented, and everything looks good to me!

**Review by VarunBhavya — APPROVED** [review]
Looks good and I approve the pull request.

**Pull request merged**